### PR TITLE
Fix obs-browser

### LINKED ISSUE #126410: OBS v27 segfaults on start
(Upstream bug, autoclosed due to NixOS not being supported: https://github.com/obsproject/obs-studio/issues/4852)

I get the following segv on start: 
```
#0  0x00007fff9f9fe510 in base::internal::SequenceLocalStorageMap::Get(int) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#1  0x00007fff9fa54795 in mojo::internal::MessageDispatchContext::MessageDispatchContext(mojo::Message*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#2  0x00007fffa065ffe4 in service_manager::mojom::ServiceManagerListenerStubDispatch::Accept(service_manager::mojom::ServiceManagerListener*, mojo::Message*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#3  0x00007fff9fa4ceb8 in mojo::internal::MultiplexRouter::ProcessIncomingMessage(mojo::internal::MultiplexRouter::MessageWrapper*, mojo::internal::MultiplexRouter::ClientCallBehavior, base::SequencedTaskRunner*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#4  0x00007fff9fa4c8e7 in mojo::internal::MultiplexRouter::Accept(mojo::Message*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#5  0x00007fff9fa46921 in mojo::Connector::DispatchMessage(mojo::Message) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#6  0x00007fff9fa4703b in mojo::Connector::ReadAllAvailableMessages() ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#7  0x00007fff9fa5a425 in mojo::SimpleWatcher::OnHandleReady(int, unsigned int, mojo::HandleSignalsState const&) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#8  0x00007fff9f9d80bd in base::TaskAnnotator::RunTask(char const*, base::PendingTask*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#9  0x00007fff9f9e744c in base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*, bool*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#10 0x00007fff9f9e7899 in non-virtual thunk to base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() () from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#11 0x00007fff9f9a85ff in base::(anonymous namespace)::WorkSourceDispatch(_GSource*, int (*)(void*), void*) ()
   from /nix/store/liyzbyv56hvad5hzm04lsl004piax9aq-obs-studio-27.0.0//lib/obs-plugins/libcef.so
#12 0x00007fffefe415ab in g_main_context_dispatch ()
   from /nix/store/dcb3cyba5wl6qimv6vwdpbi0kg0g1nlb-glib-2.68.2/lib/libglib-2.0.so.0
#13 0x00007fffefe41858 in g_main_context_iterate.constprop ()
   from /nix/store/dcb3cyba5wl6qimv6vwdpbi0kg0g1nlb-glib-2.68.2/lib/libglib-2.0.so.0
#14 0x00007fffefe4190f in g_main_context_iteration ()
   from /nix/store/dcb3cyba5wl6qimv6vwdpbi0kg0g1nlb-glib-2.68.2/lib/libglib-2.0.so.0
#15 0x00007ffff555f520 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /nix/store/z8s9ggc72xkhv9xfxk88q08rrylcxqjf-qtbase-5.15.2/lib/libQt5Core.so.5
#16 0x00007ffff5505d7b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /nix/store/z8s9ggc72xkhv9xfxk88q08rrylcxqjf-qtbase-5.15.2/lib/libQt5Core.so.5
#17 0x00007ffff550e360 in QCoreApplication::exec() ()
   from /nix/store/z8s9ggc72xkhv9xfxk88q08rrylcxqjf-qtbase-5.15.2/lib/libQt5Core.so.5
#18 0x0000000000482306 in main ()
```

This behavior was not present pre-v27.

@deviant @jb55 @MP2E

### PR DESCRIPTION
**Pull request by deviant**
Updates CEF to the latest version, and applies a couple of patches to make OBS build against that.

###### Motivation for this change

Fixes #126410

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/obs-studio/default.nix [hunk 1]
```diff
@@ -53,6 +53,14 @@ in mkDerivation rec {
     fetchSubmodules = true;
   };
 
+  patches = [
+    # Lets obs-browser build against CEF 90.1.0+
```
Comment (jb55) [inline pkgs/applications/video/obs-studio/default.nix:57]: can we upstream these patches and add a link to them here?

Comment (deviant) [reply pkgs/applications/video/obs-studio/default.nix:57]: I tried, they weren't very receptive. Maintainers seem to be of the "we don't accept changes unless you're using specifically an officially supported version of our dependencies" type which is fair enough from a maintenance perspective (especially given that they're probably dealing with gamers most of the time lol) but not one I really wish to support in nixpkgs given that we're dealing with a browser engine here. See https://github.com/obsproject/obs-browser/issues/296 and https://github.com/obsproject/obs-browser/pull/297

Comment (deviant) [reply pkgs/applications/video/obs-studio/default.nix:57]: (marking as resolved as I don't expect any luck convincing them otherwise)

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Pull request merged**